sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/download/manager/DownloadManagerRequestTask.java

```diff
@@ -45,11 +45,14 @@ protected Void doInBackground(Void... params) {
             request.setVisibleInDownloadsUi(false);
         }
         long enqueueTime = System.currentTimeMillis();
-        long downloadId = downloadManager.enqueue(request);
-        if (isCancelled()) {
-            return null;
+        try {
+            long downloadId = downloadManager.enqueue(request);
```

Comment:
Since the IllegalArgumentException is happening at this line:
```
Caused by java.lang.IllegalArgumentException: Unknown URL content://downloads/my_downloads
       at android.content.ContentResolver.insert(ContentResolver.java:1599)
       at android.app.DownloadManager.enqueue(DownloadManager.java:1163)
```
probably it makes sense to wrap with try/catch this line only?

Reply:
I'd rather not - I kept it that way intentionally (`downloadId` initialization before usage). Other way would be re-assignment, but that's probably a bad idea.